examples/using-typescript/tsconfig.json

```diff
@@ -2,7 +2,7 @@
   "include": ["./src/**/*"],
   "compilerOptions": {
     "target": "esnext",
-    "module": "commonjs",
+    "module": "esnext",
```

Comment:
TypeScript compiler throws unless this is `commonjs` on Gatsby as of 2.23.4 (current release). Given this example is intended to help individuals get started with TypeScript the current setting leads to a poor DX:

![Screen Capture_select-area_20200618214101](https://user-images.githubusercontent.com/65664227/85027511-99bccd00-b1ac-11ea-9a86-df8ab5626834.png)

Reply:
Hi @balibebas! would you be able to open a new issue for this?